### ISSUE TITLE
umap-js throws error if imported in a web worker

### ISSUE THREAD

**Issue by r03ert0** [PAIR-code/umap-js#37]
Any idea how to use umap-js in a worker? When I try to use it I get an error (other scripts import just fine). A simple example to reproduce the problem is:

`index.html`
```
<script>
var worker = new Worker("./worker.js");
</script>
```

`worker.js`
```
importScripts("https://unpkg.com/umap-js@1.3.2/lib/umap-js.js");
```

In the console (using Chrome in a Mac) the errors are:
```
Uncaught null
Script error.
```

**Comment by cannoneyed** [comment]
OK I've played around with this a bit and it's most likely because Chrome doesn't allow loading remote worker scripts from localhost. See the below stackoverflow for a bit more context:

https://stackoverflow.com/questions/21408510/chrome-cant-load-web-worker

Here's what I did to get a local web worker running on my machine:

1. Copy [umap-js.min.js](https://unpkg.com/umap-js@1.3.2/lib/umap-js.min.js) to a local file, `umap.min.js`
1.  replace `worker.js` with:
```javascript
const window = self;
importScripts("umap.min.js");
console.log(UMAP);
```

Note the sort-of kludgy `const window = self` hack to get the module correctly loaded in the web worker... I'll make sure to fix this upstream once more canonical web worker support gets finished. Note, if you want to use UMAP in a web worker, you'll have to set up message passing between the worker and the main thread. It's pretty straightforward but for now you'll have to roll your own solution
